engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -124,13 +121,15 @@ private void setConfiguration(Configuration conf, String property, Object value)
     }
   }
 
-  public class YarnAppHandleImpl implements YarnAppLauncher.YarnAppHandle
+  public class YarnAppHandleImpl implements YarnAppLauncher.YarnAppHandle, AutoCloseable
   {
     final ApplicationId appId;
+    final YarnClient yarnClient;
```

Comment:
It is not protected, it is visible by all classes in the same package. Using private not only prevents accidental misuse, it tells other developers that there can't be any YarnClient state change that bypasses YarnAppLauncherImpl